routemaster/controllers/key_registration.rb

```diff
@@ -0,0 +1,35 @@
+require 'routemaster/controllers'
+require 'routemaster/controllers/parser'
+require 'routemaster/middleware/root_authentication'
+require 'routemaster/models/client_token'
+require 'sinatra/base'
+
+module Routemaster
+  module Controllers
+    class KeyRegistration < Sinatra::Base
+      register Parser
+
+      use Routemaster::Middleware::RootAuthentication, /^\/api_key.*/
+
+      get '/api_keys' do
+        keys = Models::ClientToken.get_all
+        halt 204 if keys.empty?
+        keys.to_json
+      end
+
+      post '/api_keys', parse: :json do
+        if !data.has_key? "service_name"
+          halt 400, "expected 'service_name' key"
+        end
+        new_key = Models::ClientToken.generate_api_key(data["service_name"])
+        status 201
+        {"new_key": new_key}.to_json
```

Comment:
To be compliant with the [RFC 7231 § 6.3.2](https://tools.ietf.org/html/rfc7231#section-6.3.2) this should really include a `Location` header with the location of the new resource.